src/transformers/generation_flax_utils.py

```diff
@@ -287,8 +345,10 @@ def greedy_search_body_fn(state):
                 model_kwargs=next_model_kwargs,
             )
 
-        # The very first prompt often has sequence length > 1, so run outside of `lax.while_loop` to comply with TPU
-        state = greedy_search_body_fn(state)
+        # for now, not accepting decoder_input_ids for seq2seq models, so we should only run the for causal models
+        if not self.config.is_encoder_decoder:
```

Comment:
Actually can be have the if statement, rather on whether `input_ids.shape > 1`? For Seq2Seq it could also be the case that `decoder_input_ids` to start generating from are provided